app/src/main/java/com/tomtom/sdk/examples/maps/ConfigurableMapActivity.kt

```diff
@@ -0,0 +1,198 @@
+        initMap()
+        initializeUIElements()
+        setTagsToViewsWithDrawable()
+    }
+
+    private fun initializeUIElements() {
+        goBackImageButtonView = binding.goBackImageButton
+    }
+
+    /**
+     * Set tags to UI elements with drawable icons (needed for testing)
+     */
+    private fun setTagsToViewsWithDrawable() {
+        goBackImageButtonView.tag = R.drawable.ic_tomtom_arrow_left
+    }
+
+    /**
+     * Displaying a map
+     *
+     * MapOptions is required to initialize the map.
+     * Use MapFragment to display a map.
+     * Optional configuration: You can further configure the map by setting various properties of the MapOptions object. You can learn more in the Map Configuration guide.
+     * The last step is adding the MapFragment to the previously created container.
+     */
+    private fun initMap() {
+        val mapOptions = MapOptions(mapKey = apiKey)
+        mapFragment = MapFragment.newInstance(mapOptions)
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.map_container, mapFragment)
+            .commit()
+        mapFragment.getMapAsync { map ->
+            tomTomMap = map
+            mapView.contentDescription = applicationContext.resources.getString(R.string.map_ready)
+            enableUserLocation()
+        }
+    }
+
+    /**
+     * The SDK provides a LocationProvider interface that is used between different modules to get location updates.
+     * This examples uses the AndroidLocationProvider.
+     * Under the hood, the engine uses Android’s system location services.
+     */
+    private fun initLocationProvider() {
+        locationProvider = AndroidLocationProvider(context = this)
+    }
+
+    /**
+     * In order to show the user’s location, the application must use the device’s location services, which requires the appropriate permissions.
+     */
+    private fun enableUserLocation() {
+        if (areLocationPermissionsGranted()) {
+            showUserLocation()
+        } else {
+            requestLocationPermission()
+        }
+    }
+
+    /**
+     * The LocationProvider itself only reports location changes. It does not interact internally with the map or navigation.
+     * Therefore, to show the user’s location on the map you have to set the LocationProvider to the TomTomMap.
+     * You also have to manually enable the location indicator.
+     * It can be configured using the LocationMarkerOptions class.
+     *
+     * Read more about user location on the map in the Showing User Location guide.
+     */
+    private fun showUserLocation() {
+        initLocationProvider()
+        locationProvider.enable()
+        // zoom to current location at city level
+        onLocationUpdateListener = OnLocationUpdateListener { location ->
+            tomTomMap.moveCamera(CameraOptions(location.position, zoom = CAMERA_ZOOM_VALUE))
+            locationProvider.removeOnLocationUpdateListener(onLocationUpdateListener)
+        }
```

Comment:
nit: you can rename the constant to avoid the comment 
```suggestion
        onLocationUpdateListener = OnLocationUpdateListener { location ->
            tomTomMap.moveCamera(CameraOptions(location.position, zoom = CAMERA_ZOOM_CITY_LEVEL))
            locationProvider.removeOnLocationUpdateListener(onLocationUpdateListener)
        }
```